Bug#148

Setnea opened the pull request:
Fixing a bug, removing mpld3 library.

dizak commented on mothulity/utilities.py:
```diff
@@ -522,16 +522,17 @@ def draw_rarefaction(input_file_name,
+    plt.savefig(buf, format="png")
+    encoded = base64.b64encode(buf.getvalue()).decode('utf-8')
+    html ='<img src=\'data:image/png;base64,{}\'>'.format(encoded)
     with open(output_file_name, "wb") as fout:
```
Change mode to ```w```

dizak commented on mothulity/utilities.py:
```diff
@@ -650,9 +651,13 @@ def draw_scatter(input_file_name,
+    plt.savefig(buf, format="png")
+    encoded = base64.b64encode(buf.getvalue()).decode('utf-8')
+    html ='<img src=\'data:image/png;base64,{}\'>'.format(encoded)
+    with open(output_file_name, "wb") as fout:
```
Change mode to ```w```

Setnea replied:
Alright, totally forgot about it!

Setnea commented:
I added a legend. I'm afraid that too many groups might make it look a little messy, but it is how it is.
I also changed the code in a way that it can be easily reused for plotting variants/groups of samples.
Have a nice weekend!

dizak commented on requirements.txt:
```diff
@@ -2,7 +2,6 @@ requests
 tqdm
 biopython
 matplotlib<=2.0.0
```
Remove version requirement

dizak commented on requirements.txt:
```diff
@@ -2,7 +2,6 @@ requests
 biopython
 matplotlib<=2.0.0
-mpld3
 numpy<=1.13.3
```
Remove version requirement

dizak commented:
@Setnea I see you are trying to solve why travis builds fails unexpectedly. Can you try commenting out the ```total``` argument?

https://github.com/dizak/mothulity/blob/c16907efc6f46aabae449a27287763ca786eda6d/mothulity/utilities.py#L443

dizak commented:
```ascii``` argument in the ```tqdm``` function can accept ```bool``` or ```str```. Can you please try with passing ```ascii=' 123456789#'```? Yes, whitespace is the first char in this ```str```
